apps/microtvm/zephyr/template_project/microtvm_api_server.py

```diff
@@ -0,0 +1,533 @@
+        extract_path = os.path.splitext(project_model_library_format_tar_path)[0]
+        with tarfile.TarFile(project_model_library_format_tar_path) as tf:
+            os.makedirs(extract_path)
+            tf.extractall(path=extract_path)
+
+        if self._is_qemu(options):
+            shutil.copytree(API_SERVER_DIR / "qemu-hack", project_dir / "qemu-hack")
+
+        # Populate CRT.
+        crt_path = project_dir / "crt"
+        crt_path.mkdir()
+        for item in self.CRT_COPY_ITEMS:
+            src_path = os.path.join(standalone_crt_dir, item)
+            dst_path = crt_path / item
+            if os.path.isdir(src_path):
+                shutil.copytree(src_path, dst_path)
+            else:
+                shutil.copy2(src_path, dst_path)
+
+        # Populate Makefile.
+        shutil.copy2(API_SERVER_DIR / "CMakeLists.txt", project_dir / "CMakeLists.txt")
+        shutil.copy2(API_SERVER_DIR / "prj.conf", project_dir / "prj.conf")
+
+        # Populate crt-config.h
+        crt_config_dir = project_dir / "crt_config"
+        crt_config_dir.mkdir()
+        shutil.copy2(API_SERVER_DIR / "crt_config" / "crt_config.h", crt_config_dir / "crt_config.h")
+
+        # Populate src/
+        src_dir = project_dir / "src"
+        src_dir.mkdir()
+        shutil.copy2(API_SERVER_DIR / "src" / "main.c", src_dir / "main.c")
+
+    def build(self, options):
+        BUILD_DIR.mkdir()
+
+        cmake_args = ["cmake", ".."]
+        if options.get("verbose"):
+            cmake_args.append("-DCMAKE_VERBOSE_MAKEFILE:BOOL=TRUE")
+
+        if options.get("zephyr_base"):
+            cmake_args.append(f"-DZEPHYR_BASE:STRING={options['zephyr_base']}")
+
+        cmake_args.append(f"-DBOARD:STRING={options['zephyr_board']}")
+
+        check_call(cmake_args, cwd=BUILD_DIR)
+
+        args = ["make", "-j2"]
+        if options.get("verbose"):
+            args.append("VERBOSE=1")
+        check_call(args, cwd=BUILD_DIR)
+
+    @classmethod
+    def _is_qemu(cls, options):
+        return "qemu" in options["zephyr_board"]
+
+    def flash(self, options):
+        if self._is_qemu(options):
+            return  # NOTE: qemu requires no flash step--it is launched from open_transport.
+
+        zephyr_board = options["zephyr_board"]
+
+        # The nRF5340DK requires an additional `nrfjprog --recover` before each flash cycle.
+        # This is because readback protection is enabled by default when this device is flashed.
+        # Otherwise, flashing may fail with an error such as the following:
+        #  ERROR: The operation attempted is unavailable due to readback protection in
+        #  ERROR: your device. Please use --recover to unlock the device.
+        if zephyr_board.startswith("nrf5340dk") and _get_flash_runner() == "nrfjprog":
+            recover_args = ["nrfjprog", "--recover"]
+            recover_args.extend(_get_nrf_device_args(options))
+            self._subprocess_env.run(recover_args, cwd=build_dir)
```

Comment:
`self._subprocess_env` is not defined. Also `SubprocessEnv` is missing in this file. Maybe copy is from zephyr.py to this file.

Reply:
`build_dir` is also undefined

Reply:
resolved these problems!